### ISSUE TITLE
Public API contains not-exported types.

### ISSUE THREAD

**Issue by jhrcek** [fjvallarino/monomer#5]
There's quite a few places in the API where Haddocks mention some type, but that type is not exposed from the available modules.

Few examples of this:
-  [EventReponse](https://hackage.haskell.org/package/monomer-1.0.0.0/docs/Monomer-Widgets-Composite.html#t:EventResponse) mentions TaskHandler, but this type is nowhere to be found in the haddocks.

- [Button](https://hackage.haskell.org/package/monomer-1.0.0.0/docs/Monomer-Widgets-Singles-Button.html#v:button_) mentions ButtonCfg, but again that type is not exposed so it's unclear what could be passed in that list.

In haddocks this manifests as the unexposed type not being rendered as link. e.g.
![Screenshot from 2021-08-12 13-19-51](https://user-images.githubusercontent.com/2716069/129188475-798dc0fa-3667-45f5-b583-6d3e6b63bb4f.png)

**Comment by fjvallarino** [comment]
@jhrcek I just published a new version with the configuration types exported, plus their documentation re-organized (it existed, it just was not visible). It's quite unfortunate I had to export some types for Haddock to generate the documentation, particularly those belonging to Composite; those types were supposed to be internal helpers and not reserve names a user would potentially want to use. Anyway, it is what it is!

Thanks for reporting this! It completely slipped past me.
